app/main.py

```diff
@@ -1,3 +1,74 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | float) -> 'Distance':
+        if isinstance(other, Distance):
+            total = self.km + other.km
+        elif isinstance(other, (int, float)):
+            total = self.km + other
+        else:
+            return NotImplemented
+        return Distance(total)
+
+    def __iadd__(self, other: int | float) -> 'Distance':
```

Comment:
```suggestion
    def __iadd__(self, other: int | float | Distance) -> 'Distance':
```
Додай анотацію, усюди де очікується екземпляр класу Distance.